src/services/getEditsForFileRename.ts

```diff
@@ -3,67 +3,93 @@ namespace ts {
-            changeTracker.replaceRangeWithText(configFile, createStringRange(oldFile, configFile), newFilePath);
-        }
-    }
+        for (const property of getTsConfigPropArray(configFile, "files")) {
+            if (!isArrayLiteralExpression(property.initializer)) continue;
 
-    interface ToUpdate {
-        readonly sourceFile: SourceFile;
-        readonly toUpdate: StringLiteralLike | FileReference;
-    }
-    function isRef(toUpdate: StringLiteralLike | FileReference): toUpdate is FileReference {
-        return "fileName" in toUpdate;
+            for (const element of property.initializer.elements) {
+                if (!isStringLiteral(element)) continue;
+
+                const updated = pathUpdater(element.text, element.text, /*isImport*/ false);
+                if (updated !== undefined) {
+                    changeTracker.replaceRangeWithText(configFile, createStringRange(element, configFile), updated);
+                }
+            }
+        }
     }
 
-    function getImportsToUpdate(program: Program, oldFilePath: string, host: LanguageServiceHost): ReadonlyArray<ToUpdate> {
-        const checker = program.getTypeChecker();
-        const result: ToUpdate[] = [];
+    function updateImports(program: Program, changeTracker: textChanges.ChangeTracker, pathUpdater: PathUpdater, host: LanguageServiceHost): void {
         for (const sourceFile of program.getSourceFiles()) {
             for (const ref of sourceFile.referencedFiles) {
-                if (!program.getSourceFileFromReference(sourceFile, ref) && resolveTripleslashReference(ref.fileName, sourceFile.fileName) === oldFilePath) {
-                    result.push({ sourceFile, toUpdate: ref });
-                }
+                const updated = pathUpdater(resolveTripleslashReference(ref.fileName, sourceFile.fileName), ref.fileName, /*isImport*/ false);
+                if (updated !== undefined) changeTracker.replaceRangeWithText(sourceFile, ref, updated);
             }
 
             for (const importStringLiteral of sourceFile.imports) {
-                // If it resolved to something already, ignore.
-                if (checker.getSymbolAtLocation(importStringLiteral)) continue;
-
                 const resolved = host.resolveModuleNames
                     ? host.getResolvedModuleWithFailedLookupLocationsFromCache && host.getResolvedModuleWithFailedLookupLocationsFromCache(importStringLiteral.text, sourceFile.fileName)
                     : program.getResolvedModuleWithFailedLookupLocationsFromCache(importStringLiteral.text, sourceFile.fileName);
-                if (resolved && contains(resolved.failedLookupLocations, oldFilePath)) {
-                    result.push({ sourceFile, toUpdate: importStringLiteral });
+                const updated = resolved && firstDefined(resolved.resolvedModule ? [resolved.resolvedModule.resolvedFileName] : resolved.failedLookupLocations, path => pathUpdater(path, importStringLiteral.text, /*isImport*/ true));

```

Comment:
I think you want to check failed lookup locations irrespective of if the resolvedFileName is present because those failedLookups take precedence over whatever was resolved ?

Reply:
If it resolved successfully:
* To something not in the old directory: We should leave it alone, right?
* To something in the old directory: Then we'll recognize that at `resolvedFileName` and we're done with no need to continue on to `failedLookupLocations`, right?

Can you think of a case where this isn't sufficient?

Reply:
Even if it resolved to something not in oldDirectory, if it failed on something in old directory thats going to change the resolution no?

Reply:
Ok ignore previous (i got confused between invalidation of resolution and change in name of the resolution), why would you want to change name of import if the changes are happening in file that's present in failed lookup locations of resolution

Reply:
That's relevant if the rename has already happened and we've already updated the program. In that case there will be lots of unresolved imports that are trying to import from oldFile, but since that doesn't exist their resolution will fail.
So we have to look through their failed lookup locations to see if they *could* have resolved to oldFile if it still existed.
Note that means that we should only look at failed lookup locations if the import wasn't already successfully resolved.

Reply:
When I have this structure:
Folders b and c are in a
There is node_modules in b where resolution in file a/b/file1.ts has failed lookup locations from a/b/node_modules/ because of say ```import {a} from "foo"``` You dont want to rename "foo" to something else right when renaming a/b to a/d ?

Reply:
Newest commit should fix this, and #24345